_posts/2016-03-01-Docker-meets-Continuous-Deployment.md

```diff
@@ -0,0 +1,179 @@
+This time we will use different runner type: **Command Line**. We can execute shell script on a build agent. Because we already installed **Docker** on our build agent we can use `docker` command in our shell script.
+ 
+### Build Image 
+
+To build image we need to execute the following command: 
+
+```bash
+docker build --tag registry.private/simple_application:%build.number% .
+```
+Docker's build command will take our `Dockerfile` and build the image tagged as `registry.private/simple_application` and version `%build.number%`. 
+Variable `%build.number%` is build-in **TeamCity** variable which contains current build number.
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/release_step_1.png" alt="Release Build Image Step" title="Release Build Image Step" class="img img-responsive style-screengrab">
+
+### Push Image to Repository
+
+The image created in the previous step exists only on the agent machine. To make the image available to others we need to store it in a repository. We can use **Docker Hub**, but in our example we use private repository available under address `repository.private`. We can execute the following command to push the image to the repository.
+
+```bash
+docker push registry.private/simple_application:%build.number%
+```
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/release_step_2.png" alt="Release Push Image to Repository Step" title="Release Push Image to Repository Step" class="img img-responsive style-screengrab">
+
+### Save Version
+
+The image is safely stored in the repository, but we need to do one more step: save the version of our image. Once again we run some shell script to generate `image.version file: 
+
+```bash
+echo %build.number% > image.version
+```
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/release_step_3.png" alt="Release Save Version Step" title="Release Save Version Step" class="img img-responsive style-screengrab">
+
+## Deploy Application
+
+In the previous step we created an image which we can use now to deploy our application. We are going to create another build configuration: **Deploy**. This build will run a Docker container on **TeamCity** agent based on the image from **Release** phase.
+
+Our build configuration must contain three elements. The first one is a trigger. We will use again **Finish Build Trigger** with dependency to **Release** build configuration.
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/deploy_trigger.png" alt="Deploy Trigger" title="Deploy Trigger" class="img img-responsive style-screengrab">
+
+The second element is a version of our image. We can obtain this information from **artifact** created by **Release** build. 
+
+<img src="/assets/img/custom/blog/2016-03-01-docker-meets-continuous-deployment/deploy_artifact.png" alt="Deploy Artifact" title="Deploy Artifact" class="img img-responsive style-screengrab">
```

Comment:
basically change all ```Artifact``` to ```Artefact``` unless it's a Docker term...

I would also use "Deploying the artefact" as a title for the image and modify the other image titles accordingly, similar to the heading suggestions